task/buildah/0.1/buildah.yaml

```diff
@@ -185,6 +193,10 @@ spec:
         BUILDAH_ARGS+=("--target=${TARGET_STAGE}")
       fi
 
+      if [ -n "${BUILD_ARGS_FILE}" ]; then
+        BUILDAH_ARGS+=("--build-arg-file=${SOURCE_CODE_DIR}/${CONTEXT}/${BUILD_ARGS_FILE}")
```

Comment:
IMO we shouldn't assume the BUILD_ARGS_FILE is relative to the context dir

```suggestion
        BUILDAH_ARGS+=("--build-arg-file=${SOURCE_CODE_DIR}/${BUILD_ARGS_FILE}")
```

Buildah only does this for the Containerfile, not for the build-arg-file

```bash
$ mkdir context
$ echo 'FROM scratch' > context/Containerfile
$ buildah build -f Containerfile context/
# works
$ echo foo=bar > context/args.conf
$ buildah build -f Containerfile --build-arg-file args.conf context /
Error: open args.conf: no such file or directory
```

Reply:
but we won't be passing just "args.conf" but full path to the context dir

build args are for component, so using context dir makes sense to me

Reply:
> build args are for component, so using context dir makes sense to me

I don't think we have any basis for this. The user may want to pass the same args to all their components or different args to each component.

I would say that making the path relative to the context dir makes things unnecessarily more confusing

```yaml
- name: path-context
  value: context
- name: build-arg-file
  value: context/args.conf

# =>
# Error: open source/context/context/args.conf: no such file or directory

# Huh? Why is Konflux saying the file doesn't exist? It exists in my repo.
```

Especially considering that buildah doesn't behave like that (doesn't consider the --build-arg-file as relative to the context dir)